### PR TITLE
Support percentage by parsing percentage to double

### PR DESCRIPTION
**Pull request by Kerwinooooo**
Thanks for contributing.

- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](https://developercertificate.org)

## Description

What was changed

`canParse()` and `parseDouble()` in *DoubleParser*
They are modified to accept the strings that contains number and end with %.
Thecorressponding issue is #204 .

## Testing

Did you add a unit test?

Yes. In *CsvReaderTest* and *DoubleParserTest*

### CODE REVIEW

File: core/src/main/java/tech/tablesaw/columns/numbers/DoubleParser.java [hunk 1]
```diff
@@ -24,9 +26,15 @@ public boolean canParse(String s) {
       return true;
     }
     try {
-      Double.parseDouble(AbstractColumnParser.remove(s, ','));
+      if (isPercent(AbstractColumnParser.remove(s, ','))) {
+        s = AbstractColumnParser.remove(s, ',');
+        NumberFormat numberFormat = NumberFormat.getPercentInstance();
```
Comment (benmccann) [inline core/src/main/java/tech/tablesaw/columns/numbers/DoubleParser.java:31]: I think you could combine this line and the next into a single line:
```
NumberFormat.getPercentInstance().parse(s);
```

As it is now, `double d` looks to be unused

Comment (Kerwinooooo) [reply core/src/main/java/tech/tablesaw/columns/numbers/DoubleParser.java]: Yes, it simplify code. I have modified it. Thx for suggestion.

File: core/src/main/java/tech/tablesaw/columns/numbers/DoubleParser.java [hunk 2]
```diff
@@ -24,9 +26,14 @@ public boolean canParse(String s) {
       return true;
     }
     try {
-      Double.parseDouble(AbstractColumnParser.remove(s, ','));
+      if (isPercent(AbstractColumnParser.remove(s, ','))) {
+        s = AbstractColumnParser.remove(s, ',');
+        Number number = NumberFormat.getPercentInstance().parse(s);
```
Comment (benmccann) [inline core/src/main/java/tech/tablesaw/columns/numbers/DoubleParser.java:31]: remove unused variable
```suggestion
        NumberFormat.getPercentInstance().parse(s);
```

**Comment by lwhite1** [comment]
@benmccann do you happen to know if the use of an appropriate locale in ReadOptions would permit reading numbers with "," as a decimal point? 
If not, that would also require number parsing changes.  If there are other common use-cases, it makes me wonder if this isn't too specialized to add as its own option. Thoughts?

**Comment by benmccann** [comment]
I'm not sure what using locale for "," as decimal separator and then handling percentages have to do with one another. Maybe I'm not understanding?

**Comment by lwhite1** [comment]
I think they're both cases typically handled in Java by DecimalFormat or NumberFormat, so any solution to the "," issue that used a DecimalFormat, would probably also solve the percentage issue.

**Comment by benmccann** [comment]
Ah, okay. I'm not sure if that's supported yet, but it sounds like a good solution to this and would be interesting to test. The majority of European countries use a comma for decimal separator such as Spain, France, Norway, and Denmark.

**Comment by lwhite1** [comment]
I checked the code and Locale is only accessed by Date/Time column parsers. I will try do a little research on number parsing with locales. 

Looking at the changes proposed here, the hard-coded reference to `AbstractColumnParser.remove(s, ',')` seems like a locale-related issue in the making. Unless, I'm missing something, I don't think it should be approved as-is.

**Comment by lwhite1** [comment]
I'm still not ready to approve this. It continues to seem too limited and specialized to get it's own option.
I checked both Pandas and R, neither provide a special hook for percentages. 

- Pandas provides an option for supplying a map of column-names to conversion functions, which it calls 'converters'
- R relies on post-import processing: using a function to convert the String column to a numeric column after the data is loaded, which as @benmccann pointed out elsewhere, we already support using the mapInto() method. 

The Pandas solution is nice and general, it can be used by any columnType to convert the data to any other type. I'm not completely sure we can do that given the added complexity of strong types and generics, but it seems like we should try - if the problem is indeed worth solving.

**Comment by lwhite1** [comment]
@benmccann I'm going to close this (and the other, older version) on the grounds of method proliferation (yes i know i'm not real consistent on that). 

What are your thoughts about making this the default behavior by modifying the column type detection for floats and doubles to consider percentages?

**Comment by benmccann** [comment]
I'd be fine with that

**Comment by lwhite1** [comment]
@Kerwinooooo If you resolve the conflicts I will merge this.  Thank you.

**Comment by emilianbold** [comment]
As long as you made a decision about percentages do comment/close on #703 too please.

**Comment by Kerwinooooo** [comment]
> @Kerwinooooo If you resolve the conflicts I will merge this. Thank you.

Conflict is resolved. Thank you.

**Pull request merged**